broker/src/main/java/org/apache/rocketmq/broker/processor/ReplyMessageProcessor.java

```diff
@@ -0,0 +1,318 @@
+            Channel channel = this.brokerController.getProducerManager().findChannel(senderId);
+            if (channel != null) {
+                msg.getProperties().put(MessageConst.PROPERTY_PUSH_REPLY_TIME, String.valueOf(System.currentTimeMillis()));
+                replyMessageRequestHeader.setProperties(MessageDecoder.messageProperties2String(msg.getProperties()));
+
+                try {
+                    RemotingCommand pushResponse = this.brokerController.getBroker2Client().callClient(channel, request);
+                    assert pushResponse != null;
+                    switch (pushResponse.getCode()) {
+                        case ResponseCode.SUCCESS: {
+                            response.setCode(ResponseCode.SUCCESS);
+                            response.setRemark(null);
+                            pushOk = true;
+                            break;
+                        }
+                        default: {
+                            response.setCode(ResponseCode.SYSTEM_ERROR);
+                            response.setRemark("push reply message to requester fail");
+                            log.warn("push reply message to <{}> return fail, remark: {}", senderId, response.getRemark());
+                        }
+                    }
+                } catch (InterruptedException e) {
+                    response.setCode(ResponseCode.SYSTEM_ERROR);
+                    response.setRemark("push reply message to requester fail");
+                    log.warn("push reply message to <{}> fail. {}", senderId, channel, e);
+                } catch (RemotingException e) {
+                    response.setCode(ResponseCode.SYSTEM_ERROR);
+                    response.setRemark("push reply message to requester fail");
+                    log.warn("push reply message to <{}> fail. {}", senderId, channel, e);
+                }
+            } else {
+                response.setCode(ResponseCode.SYSTEM_ERROR);
+                response.setRemark("channel of <" + senderId + "> not found");
+                log.warn("push reply message fial, channel of <{}> not found.", senderId);
+            }
+            return pushOk;
+        }
+        log.warn("REPLY_TO is null, can not reply message");
+        response.setCode(ResponseCode.SYSTEM_ERROR);
+        response.setRemark("REPLY_TO is null");
+        return pushOk;
+    }
+
+    private void handlePutMessageResult(PutMessageResult putMessageResult, final RemotingCommand response,
+        final RemotingCommand request, final MessageExt msg,
+        final SendMessageResponseHeader responseHeader, SendMessageContext sendMessageContext,
+        ChannelHandlerContext ctx,
+        int queueIdInt) {
+        if (putMessageResult == null) {
+            response.setRemark("push reply to requester success, but store putMessage return null");
+            return;
+        }
+        boolean sendOK = false;
+
+        switch (putMessageResult.getPutMessageStatus()) {
```

Comment:
I think the setting of response code and remark is a bit messy. When pushReplyMessage is called, response will be set, then handlePutMessageResult is called, response will be set again.